bindings/java/src/org/sleuthkit/datamodel/CaseDatabaseFactory.java

```diff
@@ -267,6 +269,23 @@ private void createArtifactTables(Statement stmt) throws SQLException {
 				+ "FOREIGN KEY(attribute_type_id) REFERENCES blackboard_attribute_types(attribute_type_id))");		
 	}
 	
+	private void createAnalysisResultsTables(Statement stmt) throws SQLException  {
+		stmt.execute("CREATE TABLE tsk_analysis_result (obj_id " + dbQueryHelper.getBigIntType() + " NOT NULL, "
+				+ "conclusion TEXT, "
+				+ "significance INTEGER NOT NULL, "
+				+ "confidence INTEGER NOT NULL, "
+				+ "configuration TEXT, justification TEXT, "
+				+ "ignore_score INTEGER DEFAULT 0, " // boolean	
+				+ "FOREIGN KEY(obj_id) REFERENCES tsk_objects(obj_id) ON DELETE CASCADE"
```

Comment:
Can this foreign key go to the artifacts table?

Reply:
I originally did have this FK refer to the blackboard_artifacts(artifact_obj_id) column.  

With SQLite, I was getting an error getting my INSERT_ANALYSIS_RESULT prepared statement, which kept failing with an error message "A problem occurred while trying to acquire a cached PreparedStatement in a background thread"

When I tried with PostGres, it gave me a more helpful message pointing out that column artifact_obj_id in the blackboard_artifacts is not UNIQUE.  When I modified the create table SQL for blackboard_artifacts to add a UNIQUE constraint to that column (perhaps we intentionally or unintentionally dropped that when we moved DB creation from C++ to Java?), it worked on PostGres.  

I will try again with SQLite and if adding the UNIQUE constraints works, I will set the FK to the blackboard_artifacts table.

Reply:
Added a UNIQUE(artifact_object_id) constraint to blackboard_artifacts table and changes the FK constraint in tsk_analysis_results to blackboard_artifacts (artifact_object_id)